api/models.py

```diff
@@ -69,3 +69,13 @@ def convert_values_to_string(cls, elements):
         for key, element in elements.items():
             converted_elements[key] = json.dumps(element)
         return converted_elements
+
+
+@receiver(models.signals.post_save, sender=FormData)
+def render_form_data_html(sender, instance, created, raw, **kwargs):
+    """Renders FormData.html after a new FormData has been created."""
+    if created and not raw:
```

Comment:
I think doing it with a check for 'created' will prevent the template from getting rerendered after a save other than the initial creation of a form.